### ISSUE TITLE
Aspiratorex better console output

### ISSUE THREAD

**Issue by agrignard** [agrignard/WhatsNext#35]
As Aspiratorex is in theory maint to be run many times, like once a week per city, it would be nice (if possible) to have a better output (int the console) or in a log to know what has been working or what didn't. Maybe with an option to make it verbose or not

Right now I have to admit that I don't really know if the aspiration went well or not and if not what went wrong.

**Comment by tnguyenh** [comment]
Yes. Yet it is not as simple as it sounds. It is also a priority problem, because integrating aspiratorex in the GUI will require to smartly process the errors. 

Technically, everything is feasible, but the behavior of console.log, asynchrony and dealing with several code files and functions requires a bit of work. So first we need to discuss about the concepts of the log, and determine what should be seen in the console, and what should be written to files, and to which files. 

I think the following:
- console output is mostly to see if aspiratorex is currently processing. In that case, we should accept that the log is a bit messy, due to asynchronous execution. Most of the mess is due to timeouts messages. If we keep them, it's very messy. If we remove them, we will see no messages during timeouts, aspiratorex will be processing without any logs. So it's a bit confusing, you don't know if aspiratorex has crashed or still processing. Doing something in between, in a more compact way, is very tricky, because each link is treated separatly in an asynchronous way.
- log files would be used to keep only useful information, and timeouts problems won't be displayed. Much easier to read, but you need to open the file to see if there were any problem. Furthermore, opening a log file won't highlight errors with colors if opened with a basic text editor (this is what data would do ?). Not a big deal ? Anyway it could be display after the timeout logs in the console...

Errors are of different kinds. There are downloads errors, which are expected to happen often. That should appear in the log. What about other errors, like IO errors, system errors ? It's easy to keep them in the console, but lost in the messy messages for timeouts. More difficult to keep them in a log due to the complexity of the code (and messy). In that case, the function console.error should be overriden.

Right now, I would see it this way: 
- keep a messy output in the console, in order to see the process flow of aspiratorex
- save outcomes and errors in logs.

All of this will be dealt keeping in mind that logs outputs should be designed at the same time for execution in the console and in the GUI. To that matter, there is a lot of questions on how to do that (for example, keep logs/error logs in webSources for each venue separatly ?).

I will be on it very soon, because I will start soon the integration of aspiratorex/scrapex in the GUI.
